src/app/pages/deprecation/DeprecationAction.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace app\pages\deprecation;
+
+use Slim\Psr7\Request;
+use Slim\Views\Twig;
+use app\domain\Version;
+use app\domain\util\Redirect;
+use app\domain\Artifact;
+
+class DeprecationAction
+{
+  private Twig $view;
+
+  public function __construct(Twig $view)
+  {
+    $this->view = $view;
+  }
+
+  public function __invoke(Request $request, $response, $args)
+  {
+    return $this->view->render($response, 'deprecation/deprecation.twig', [
+      'versions' => self::getVersions(),
+	  'features' => self::getFeatures()
+    ]);
+  }
+  
+  private static function getVersions(): array
+  {
+    return ["3", "4", "5", "6", "7", "8", "9.1", "9.2", "9.3"];
+  }
+  
+  private static function getFeatures(): array
+  {
+	return [
+	  new Feature(
+	    "CMS Page", 
+		"/doc/8.0/designer-guide/user-interface/web-page/html-content-in-the-cms.html",
+		"Html Dialog", 
+		"/doc/latest/designer-guide/user-interface/user-dialogs/html-dialogs.html",
+		"3",
+		"8",
+		"9.3"),		
+	  new Feature(
+	    "Macros in RTF documents", 
+		"/doc/8.0/designer-guide/user-interface/web-page/index.html",
+		"Doc Factory", 
+		"/market/doc-factory",
+		"3",
+		"8",
+		"9.3"),		
+	  new Feature(
+	    "Import Ivy 3.9 Project", 
+		"/doc/7.0/DesignerGuideHtml/ivy.processmodeling.html#ivy-project-convert39",
+		"", 
+		"",
```

Comment:
looks as if we need a simplified constructor: 
I'd assume that must parameters are 'Feature Name, Introduce version, Deprecate Version' others may be ommitted.
If we would have these params, I think the readability of this code would be improved as well.

Reply:
I'm a php newby. Will discuss with Alex tomorrow how to improve. We could also make a *.json file per feature and read the information from those files. @alexsuter what is the best solution here?

Reply:
yes. If you like you can add json. It's pretty simply in PHP to parse a json. Have a look at the TeamsAction.
You even do not need to model out the Feature. You can access everything from the json object without a model. if you like.